File: AndroidXPatternsApp/jmvvm-app/src/main/java/com/woowrale/jmvvm/ui/base/BaseActivity.java
package com.woowrale.jmvvm.ui.base;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.woowrale.jmvvm.MainApplication;
import com.woowrale.jmvvm.di.components.MainComponent;

public abstract class BaseActivity extends AppCompatActivity implements BasePresenter.View {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getDaggerMainComponent().inject(this);
    }

    protected MainComponent getDaggerMainComponent() {
        return ((MainApplication) getApplication()).getMainComponent();
    }

    protected abstract void initDagger();

}

File: AndroidXPatternsApp/kmvvm-app/src/main/java/com/woowrale/kmvvm/ui/base/BaseActivity.kt
package com.woowrale.kmvvm.ui.base

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.woowrale.kmvvm.MainApplication
import com.woowrale.kmvvm.di.components.MainComponent

abstract class BaseActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        getDaggerMainComponent().inject(this)
    }

    protected fun getDaggerMainComponent(): MainComponent {
        val mainApplication = MainApplication()
        mainApplication.onCreate()
        return mainApplication.mainComponent
    }

    protected abstract fun initDagger()
}
